MathlibTest/CommandStart.lean

```diff
@@ -0,0 +1,332 @@
+
+This part of the code
+  'Nat]  [Add'
+should be written as
+  'Nat] [Add'
+
+note: this linter can be disabled with `set_option linter.style.commandStart false`
+-/
+#guard_msgs in
+omit  [h : Add Nat]  [Add Nat]
+end
+
+-- Code inside `run_cmd` is not checked at all.
+run_cmd
+  for _ in [0] do
+    let _ ← `(
+      end)
+
+def Card : Type → Nat := fun _ => 0
+
+/-- Symbols for use by all kinds of grammars. -/
+inductive Symbol (T N : Type)
+  /-- Terminal symbols (of the same type as the language) -/
+  | terminal (t : T) : Symbol T N
+  /-- Nonterminal symbols (must not be present when the word being generated is finalized) -/
+  | nonterminal (n : N) : Symbol T N
+deriving
+  DecidableEq, Repr
+
+-- embedded comments do not cause problems!
+#guard_msgs in
+open Nat in -- hi
+example : True := trivial
+
+-- embedded comments do not cause problems!
+#guard_msgs in
+open Nat in
+ -- hi
+example : True := trivial
+
+-- embedded comments do not cause problems!
+#guard_msgs in
+open Nat in
+  -- hi
+example : True := trivial
+
+structure X where
+  /-- A doc -/
+  x : Nat
+
+open Nat in /- hi -/
+example : True := trivial
+
+-- The notation `0::[]` disables the linter
+variable (h : 0::[] = [])
+
+/--
+warning: 'section' starts on column 1, but all commands should start at the beginning of the line.
+note: this linter can be disabled with `set_option linter.style.commandStart false`
+-/
+#guard_msgs in
+ section
+
+/--
+warning: extra space in the source
+
+This part of the code
+  'example    : True'
+should be written as
+  'example : True'
+
+note: this linter can be disabled with `set_option linter.style.commandStart false`
+-/
+#guard_msgs in
+example    : True := trivial
+
+-- Additional spaces after the colon are not linted yet.
+#guard_msgs in
+example :     True := trivial
+
+/-- A doc string -/
+-- comment
+example : True := trivial
+
+/--
+warning: extra space in the source
+
+This part of the code
+  'example  :  True'
+should be written as
+  'example : True'
+
+note: this linter can be disabled with `set_option linter.style.commandStart false`
+-/
+#guard_msgs in
+example  :  True :=trivial
+
+/--
+warning: missing space in the source
+
+This part of the code
+  '(a: Nat)'
+should be written as
+  '(a : Nat)'
+
+note: this linter can be disabled with `set_option linter.style.commandStart false`
+-/
+#guard_msgs in
+variable (a: Nat)
+
+/--
+warning: missing space in the source
+
+This part of the code
+  '(_a: Nat)'
+should be written as
+  '(_a : Nat)'
+
+note: this linter can be disabled with `set_option linter.style.commandStart false`
+-/
+#guard_msgs in
+example (_a: Nat) : True := trivial
+
+/--
+warning: missing space in the source
+
+This part of the code
+  '{a: Nat}'
+should be written as
+  '{a : Nat}'
+
+note: this linter can be disabled with `set_option linter.style.commandStart false`
+-/
+#guard_msgs in
+example {a: Nat} : a = a := rfl
+
+set_option linter.style.commandStart.verbose true in
```

Comment:
Note that this does not do anything, and the output in the command is no longer produced: remove this set_option and the comment?

Reply:
What do you think about this?